app/models/vm/operations/power.rb

```diff
@@ -7,10 +7,27 @@ module Vm::Operations::Power
     api_relay_method :suspend
 
     supports :suspend do
-      msg = unsupported_reason(:control) unless supports_control?
-      msg ||= _('The VM is not powered on') unless vm_powered_on?
+      msg = supports_control_powered_on
```

Comment:
@durandom - If we are going to refactor all the VM operations we will end up with a lot of duplicated code here and throughout the provider code, I prefer we address that, Im open to other suggestions if this is not perfect.